### ISSUE TITLE
Support develop-branches / releases of 3rd-party modules

### ISSUE THREAD

**Issue by looooo** [FreeCAD/FreeCAD-addons#151]
Would be nice to somehow support different version of 3rd-party modules. I guess this would be useful for:
1. 3rd-party modules dependent on specific versions of freecad. In this case the addon-manager should install a version of the 3rd-party module which works with the version of freecad.
2. Install and test specific features which are developed in a seperate branch (keeps master-branch stable, avoids backward incompatibilities.)

Idea came up in this issue:
https://github.com/looooo/freecad.gears/issues/1

**Comment by sgrogan** [comment]
Interesting.
For 2, it seems possible to do manually.
I tested with the Gears WB that was install with the add-on manager using git-python.
I bashed into %appdata%/FreeCAD/Mod/FCGear
I was able to git checkout develop
I was also able to git checkout -b test
I changed the version in pygears/_init_.py, and after a FreeCAD re-start the change is picked up in the FreeCAD python console "print(pygears.__version__)"

I guess the challenge would be all the visual stuff so the user knows they are not at HEAD of master or on a branch (maybe also if the branch is local, not in upstream)
